packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -220,15 +220,8 @@ private IEnumerable<MethodBodyStatement> AppendHeaderParameters(HttpRequestApi r
                 string? format;
                 ValueExpression valueExpression;
                 GetParamInfo(paramMap, operation, inputParameter, out type, out format, out valueExpression);
-                ValueExpression toStringExpression;
-                if (type == null || type.IsLiteral || type.Equals(typeof(string)))
-                {
-                    toStringExpression = valueExpression;
-                }
-                else
-                {
-                    toStringExpression = TypeFormattersSnippets.ConvertToString(valueExpression, Literal(format));
-                }
+                var convertToStringExpression = TypeFormattersSnippets.ConvertToString(valueExpression, Literal(format));
+                ValueExpression toStringExpression = type?.Equals(typeof(string)) == true ? valueExpression : convertToStringExpression;
```

Comment:
I think it would be simpler if we just combine them together like:
```suggestion
                ValueExpression toStringExpression = type?.Equals(typeof(string)) == true ?
                	valueExpression : 
                	TypeFormattersSnippets.ConvertToString(valueExpression, Literal(format));
```